main.py

```diff
@@ -0,0 +1,168 @@
+
+    # 1. We collect all events objects from the query time span
+    # and group them by day.
+
+    # keep everything in RAM, keyed by day
+    events = {}
+    count = 0
+    for event in fetch_events():
+        event_datetime = datetime.fromtimestamp(event["created"])
+        daystring = event_datetime.strftime("%Y-%m-%d")
+        if daystring == nowstring:
+            continue
+        if daystring not in events:
+            events[daystring] = []
+        count += 1
+        events[daystring].append(event)
+    print("Fetched %d events." % count)
+
+    # 2. We go through day by day and see where new entries have been added
+    # since the last run. For these we create a backup file.
+
+    redis_retention = REDIS_RETENTION_DAYS * 24 * 60 * 60
+
+    for daystring in sorted(events.keys()):
+        print("Processing day %s with %d events" % (daystring, len(events[daystring])))
+
+        # possibly skip days already archived
+        redis_value = redis.get(daystring)
+        if redis_value is not None:
+            num_entries_before = int(redis_value)
+            if len(events[daystring]) <= num_entries_before:
+                print("Skipping day %s, already archived" % daystring)
+                continue
+
+        try:
+            dt = datetime.strptime(daystring, "%Y-%m-%d")
+
+            filename = dump_to_file(daystring, events[daystring])
+            target_path = ""
+            if S3_PATH is not None:
+                target_path = dt.strftime(S3_PATH)
+            target_path += "/" + filename
+            upload_backup(s3, filename, target_path)
+            os.remove(filename)
+
+            # write number of entries per day to redis
+            redis.setex(daystring, redis_retention, str(len(events[daystring])))
+        except Exception as ex:
+            sys.stderr.write("ERROR: No backup created for %s\n" % daystring)
+            sys.stderr.write(traceback.format_exc() + "\n")
+
+    del redis
+    del s3
+    del events
+
+    print("Done for today.")
+
+
+def fetch_events():
+    has_more = True
+    params = {"limit": 100}
+    auth = HTTPBasicAuth(STRIPE_API_KEY, '')
+    while has_more:
+        r = requests.get("https://api.stripe.com/v1/events",
+                         params=params,
+                         auth=auth)
+        r.raise_for_status()
+        j = r.json()
+        has_more = j["has_more"]
+        params["starting_after"] = j["data"][-1]["id"]
+        for item in j["data"]:
+            yield item
+
+
+def dump_to_file(daystring, events):
+    """
+    Creates a backup file from a list of events
+    """
+    dump = ""
+    hasher = hashlib.sha1()
+    for item in events:
+        j = json.dumps(item, sort_keys=True) + "\n"
+        hasher.update(j)
+        dump += j
+    sha1hash = hasher.hexdigest()
+    filename = daystring + "_" + sha1hash[0:6] + ".jsonl.aes-256-cbc"
+
+    # encrypt and write to file
+    with open(filename, 'w') as dumpfile:
```

Comment:
Since you are still saving the encrypted file. How about at least writing to a tmpfs (docker has a `--tmpfs` options) and making sure the files are `0600`. This means no other user can read the files and the files are still only in memory and gone if the container stops.